x-pack/plugins/security_solution/server/lib/telemetry/endpoint.test.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { savedObjectsRepositoryMock } from 'src/core/server/mocks';
+import {
+  mockFleetObjectsResponse,
+  mockFleetEventsObjectsResponse,
+  MockOSFullName,
+  MockOSPlatform,
+  MockOSVersion,
+} from './endpoint.mocks';
+import { ISavedObjectsRepository, SavedObjectsFindResponse } from 'src/core/server';
+import { AgentEventSOAttributes } from './../../../../ingest_manager/common/types/models/agent';
+import { Agent } from '../../../../ingest_manager/common';
+import * as endpointTelemetry from './endpoint';
+import * as fleetSavedObjects from './fleet_saved_objects';
+
+describe('test security solution endpoint telemetry', () => {
+  let mockSavedObjectsRepository: jest.Mocked<ISavedObjectsRepository>;
+  let getFleetSavedObjectsMetadataSpy: jest.SpyInstance<Promise<SavedObjectsFindResponse<Agent>>>;
```

Comment:
:grey_question: These jest.Mocked and jest.SpyInstance deals look pretty cool. Good docs for them anywhere I can read?